packages/embark-transaction-manager/src/lib/eth_sendTransaction.ts

```diff
@@ -0,0 +1,87 @@
+import async from "async";
+import { Callback, Embark, Events, Logger } /* supplied by @types/embark in packages/embark-typings */ from "embark";
+import { __ } from "embark-i18n";
+import Web3 from "web3";
+const { blockchain: blockchainConstants } = require("embark-core/constants");
+import { EventEmitter } from "events";
+import Transaction from "./Transaction";
+
+export default class EthSendTransaction implements Transaction {
+  public readonly events: Events;
+  public readonly logger: Logger;
+  private signTransactionQueue: any;
+  private nonceCache: any = {};
+  constructor(readonly embark: Embark, readonly web3: Web3, public nodeAccounts: any[], public accounts: any[], readonly txEvents: EventEmitter) {
+    this.events = embark.events;
+    this.logger = embark.logger;
+
+    if (accounts.length) {
+      embark.registerActionForEvent("blockchain:proxy:request", this.checkRequestFor_eth_sendTransaction.bind(this));
+    }
+
+    // Allow to run transaction in parallel by resolving the nonce manually.
+    // For each transaction, resolve the nonce by taking the max of current transaction count and the cache we keep locally.
+    // Update the nonce and sign it
+    this.signTransactionQueue = async.queue(({ payload, account }, callback: Callback<any>) => {
```

Comment:
Might as well include the transactionQueue in the `if` or reverse the `if` and return?

Reply:
Updated.